apollo-backend/src/test/java/io/logz/apollo/UndeployedServicesTest.java

```diff
@@ -0,0 +1,137 @@
+package io.logz.apollo;
+
+import io.logz.apollo.clients.ApolloTestClient;
+import io.logz.apollo.dao.DeployableVersionDao;
+import io.logz.apollo.dao.DeploymentDao;
+import io.logz.apollo.dao.GroupDao;
+import io.logz.apollo.dao.ServiceDao;
+import io.logz.apollo.exceptions.ApolloClientException;
+import io.logz.apollo.helpers.Common;
+import io.logz.apollo.helpers.ModelsGenerator;
+import io.logz.apollo.helpers.StandaloneApollo;
+import io.logz.apollo.models.DeployableVersion;
+import io.logz.apollo.models.Deployment;
+import io.logz.apollo.models.Environment;
+import io.logz.apollo.models.EnvironmentServices;
+import io.logz.apollo.models.Group;
+import io.logz.apollo.models.Service;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+public class UndeployedServicesTest {
```

Comment:
I want us to sit on this test and write it again together when you are done with the rest please